curve25519-dalek/src/lemmas/common_lemmas/bit_lemmas.rs

```diff
@@ -35,6 +35,15 @@ pub proof fn lemma_bit_or_is_plus(a: u64, b: u64, k: u64)
 {
 }
 
+pub proof fn lemma_bit_or_is_plus_overflow(a: u64, b: u64, k: u64)
+    by (bit_vector)
+    requires
+        a < 1u64 << k,
+    ensures
+        a | (b << k) == a + (b << k),
+{
+}
+
```

Comment:
No need to add a new function, this generalizes `lemma_bit_or_is_plus`, so just remove the extra requires there.